### PR TITLE
Improvements: allowed actions support on FE

### PR DESCRIPTION
**Pull request by lciolecki**
It's next part of changes from https://github.com/TouK/nussknacker/pull/1401. 
* supporting allowed actions on FE 
* some cleanup
* add unarchived button

### CODE REVIEW

File: ui/client/components/toolbars/process/buttons/ArchiveButton.tsx [hunk 1]
```diff
@@ -41,7 +41,7 @@ function ArchiveButton(props: StateProps) {
 const mapState = (state: RootState) => {
   return {
     processId: getProcessId(state),
-    isRunning: isRunning(state),
+    canArchive: !isSubprocess(state) && isArchivePossible(state) || isSubprocess(state),
```
Comment (mproch) [inline ui/client/components/toolbars/process/buttons/ArchiveButton.tsx:44]: let's add parentheses here... or can we just write:
 isArchivePossible(state) || isSubprocess(state)??

File: ui/client/components/toolbars/process/buttons/UnArchiveButton.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  return (
+    <ToolbarButton
+      name={t("panels.actions.process-unarchive.button", "unarchive")}
+      icon={<Icon/>}
+      disabled={!isArchived}
+      onClick={() => isArchived && toggleConfirmDialog(
+        true,
+        DialogMessages.unArchiveProcess(processId),
+        () => HttpService.unArchiveProcess(processId).then(() => history.push(redirectPath)),
+        t("panels.actions.process-archive.yes", "Yes"),
```
Comment (mproch) [inline ui/client/components/toolbars/process/buttons/UnArchiveButton.tsx:34]: Why two lines above we have DialogMessages.unArchiveProcess, and here we inline?

File: ui/client/components/toolbars/process/buttons/ArchiveButton.tsx (hunk 1 above)
Comment (lciolecki) [reply ui/client/components/toolbars/process/buttons/ArchiveButton.tsx:44]: We can't because subprocess can't get process state..

Comment (mproch) [reply ui/client/components/toolbars/process/buttons/ArchiveButton.tsx:44]: so at least  isSubprocess(state) || isArchivePossible(state)
- it's equivalent, isn't it?

**Review by mproch — APPROVED** [review]

**Pull request merged**